src/components/fields/checkbox-group/checkbox-group.tsx

```diff
@@ -0,0 +1,98 @@
+import { Form } from "@lifesg/react-design-system/form";
+import { without } from "lodash";
+import React, { useEffect, useState } from "react";
+import * as Yup from "yup";
+import { useValidationSchema } from "../../../utils/hooks";
+import { IGenericFieldProps } from "../../frontend-engine";
+import { Label, StyledCheckbox } from "./checkbox-group.styles";
+import { ICheckboxGroupSchema } from "./types";
+
+export const CheckboxGroup = (props: IGenericFieldProps<ICheckboxGroupSchema>) => {
+	// =============================================================================
+	// CONST, STATE, REFS
+	// =============================================================================
+	const {
+		schema: { label, options, validation, ...otherSchema },
+		id,
+		value,
+		error,
+		onChange,
+	} = props;
+
+	const [stateValue, setStateValue] = useState<string[]>(value || []);
+	const { setFieldValidationConfig } = useValidationSchema();
+
+	// =============================================================================
+	// EFFECTS
+	// =============================================================================
+	useEffect(() => {
+		setFieldValidationConfig(
+			id,
+			Yup.array()
+				.of(Yup.string())
+				.test("is-empty-array", "An option is required", (value) => {
+					if (!value) return true;
+
+					return value.length > 0;
```

Comment:
need to check whether field is required first because this rule should only be applicable if the field is required